whois-api/src/main/java/net/ripe/db/whois/api/rdap/RdapCrossOriginFilter.java

```diff
@@ -0,0 +1,40 @@
+package net.ripe.db.whois.api.rdap;
+
+import com.google.common.net.HttpHeaders;
+import org.eclipse.jetty.servlets.CrossOriginFilter;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.ws.rs.HttpMethod;
+import java.io.IOException;
+
+/**
+ * RDAP specific changes to Cross-Origin Resource Sharing (CORS).
+ *
+ *   "When responding to queries, it is RECOMMENDED that servers use the
+ *    Access-Control-Allow-Origin header field, as specified by
+ *    [W3C.REC-cors-20140116].  A value of "*" is suitable when RDAP is
+ *    used for public resources."
+ *
+ * Ref. https://tools.ietf.org/html/rfc7480#section-5.6
+ *
+ */
+public class RdapCrossOriginFilter extends CrossOriginFilter {
+
+    @Override
+    public void doFilter(final ServletRequest request, final ServletResponse response, final FilterChain chain) throws IOException, ServletException
+    {
+        handle((HttpServletRequest)request, (HttpServletResponse)response);
+        super.doFilter(request, response, chain);
+    }
+
+    private void handle(final HttpServletRequest request, final HttpServletResponse response) {
+        if ((request.getHeader(HttpHeaders.ORIGIN) == null) && request.getMethod().equals(HttpMethod.GET)) {
```

Comment:
question: why is the requirement for the ORIGIN to be null placed? I do not see anything that cann point to this in the rfc...mind clarifying?

Reply:
The RFC recommends that the CORS header *always* appears in the GET response. If there is an Origin header, then the response is handled by the existing CrossOrigin filter. This PR adds a CORS header if there *isn't* an Origin header in the request.